libursa/src/signatures/ed25519.rs

```diff
@@ -34,6 +34,36 @@ impl Ed25519Sha512 {
             ))),
         }
     }
+
+    pub fn sign_key_to_key_exchange(sk: &PrivateKey) -> Result<PrivateKey, CryptoError> {
+        // Length is normally 64 but we only need the secret from the first half
+        if sk.len() < 32 {
+            return Err(CryptoError::ParseError(format!(
+                "Invalid private key provided"
+            )));
+        }
+        // hash secret
+        let hash = sha2::Sha512::digest(&sk[..32]);
+        let mut output = [0u8; 32];
+        output.copy_from_slice(&hash[..32]);
+        // clamp result
+        let secret = x25519_dalek::StaticSecret::from(output);
+        Ok(PrivateKey(secret.to_bytes().to_vec()))
+    }
+
```

Comment:
Please add comments here to document the purpose of the expand_keypair function and how to use it